cmd/harmony/main.go

```diff
@@ -245,28 +257,32 @@ func main() {
 	flag.Parse()
 
 	initSetup()
+	var currentNode *node.Node
+	var consensus *consensus.Consensus
 	nodeConfig := createGlobalConfig()
-
-	// Init logging.
-	loggingInit(*logFolder, nodeConfig.StringRole, *ip, *port, *onlyLogTps)
-
-	// Start Profiler for leader if profile argument is on
-	if nodeConfig.StringRole == "leader" && (*profile || *metricsReportURL != "") {
-		prof := profiler.GetProfiler()
-		prof.Config(nodeConfig.ShardIDString, *metricsReportURL)
-		if *profile {
-			prof.Start()
+	if *isArchival {
+		currentNode, nodeConfig = setupArchivalNode(nodeConfig)
+		loggingInit(*logFolder, nodeConfig.StringRole, *ip, *port, *onlyLogTps)
+		log.Info("New Harmony Node ====", "Role", currentNode.NodeConfig.Role(), "multiaddress", fmt.Sprintf("/ip4/%s/tcp/%s/p2p/%s", *ip, *port, nodeConfig.Host.GetID().Pretty()))
+		go currentNode.DoBeaconSyncing()
```

Comment:
why archival node do beaconsyncing only?  even though we only support beacon chain right now, the function should be generic or wrap around do beacon syncing.

Reply:
I would suggest using MainDB to store the backup instead of the BeaconDB. This unifies the storage part.  But it needs some modification from either beaconsync or statesync

Reply:
As discussed offline, we will fix this once you merge https://github.com/harmony-one/harmony/pull/598/